### PR TITLE
Add Rounded Corners

### PR DESCRIPTION
**Pull request by johnny-duo**
This PR resolves issue #1523: 

https://user-images.githubusercontent.com/108750639/199562241-4510d730-42a7-4466-a78e-99f8efd88a84.mp4

### CODE REVIEW

**Comment by calda** [comment]
Thank you very much for working on this! 

When the implementation is done and ready for review, could you also add a test case for this to the regression suite? You'll just need to:
 1. add the animation json file to the `Tests/Samples`, ideally as `Tests/Samples/Issues/issue_1523.json`
 2. run the unit test target, either in Xcode or by running `bundle exec rake test:package`

The code will also need to be formatted according to [Airbnb's Swift Style Guide](https://github.com/airbnb/swift). You can do this automatically by running `bundle exec rake format:swift`.

Thanks!

**Comment by calda** [comment]
I also see you've only implemented this feature for the Main Thread rendering engine and not the new Core Animation rendering engine. That's fine for this PR -- I'll port this to the Core Animation rendering engine in a follow-up.

**Comment by johnny-duo** [comment]
@calda Hi! Thanks for looking at this PR! I'm having some issues with snapshot tests in LottieFiles/gradient_shapes and TypeFace/Apostrophe. It looks like they have "ty": "rd" inside of the jsons so I think this is expected.

Is it possible to view the Snapshot results?

**Comment by johnny-duo** [comment]
Yup! This PR is only for the main thread engine for now.

File: Sources/Private/MainThread/NodeRenderSystem/Nodes/ModifierNodes/RoundedCornersNode.swift [hunk 1]
```diff
@@ -0,0 +1,81 @@
+//
+//  RoundedCornersNode.swift
+//  Lottie
+//
+//  Created by Duolingo on 10/31/22.
```
Comment (calda) [inline Sources/Private/MainThread/NodeRenderSystem/Nodes/ModifierNodes/RoundedCornersNode.swift:5]: You work at Duolingo? Cool!

I have a 1083 day streak learning Spanish 😄 Really love what Duolingo has been doing with Lottie lately -- cutting edge stuff. The new feature where the characters lip-sync with the text-to-speech synth honestly blew my mind. Is that using Lottie or some sort of custom lip sync renderer?

Comment (johnny-duo) [reply Sources/Private/MainThread/NodeRenderSystem/Nodes/ModifierNodes/RoundedCornersNode.swift:5]: Yeah! Glad you're enjoying that! I implemented that on iOS actually :D. We couldn't figure out a way to do that with Lottie so we ended up having to use a different more state-machine based animation Library

Comment (calda) [reply Sources/Private/MainThread/NodeRenderSystem/Nodes/ModifierNodes/RoundedCornersNode.swift:5]: So cool, awesome work! (Not surprised that isn't possible with Lottie)

Are you on Twitter? Would love to give you a follow if so -- I'm https://twitter.com/calstephens98

**Comment by calda** [comment]
> Is it possible to view the Snapshot results?

Yes, there are a few ways to do this:

 1. Run the snapshot tests in Xcode. The snapshots are added as attachments to the test results in the Test navigator.
 2. Add `isRecording = true` to the top of `testMainThreadRenderingEngine` in `SnapshotTests.swift` and re-run the tests. This updates the on-disk snapshots (which you'll need to do to make the tests pass, if this affects existing animations).

File: Sources/Private/Utility/Primitives/BezierPathRoundExtension.swift [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        path.roundCorners(radius: radius))
+    }
+
+    return CompoundBezierPath(paths: newPaths)
+  }
+}
+
+extension BezierPath {
+  // Round corners of a single bezier
+  func roundCorners(radius: CGFloat) -> BezierPath {
```
Comment (calda) [inline Sources/Private/Utility/Primitives/BezierPathRoundExtension.swift:46]: Thank you for implementing this algorithm in Swift!

**Review by calda — APPROVED** [review]

**Review by calda — APPROVED** [review]
Thank you!

File: Sources/Private/MainThread/NodeRenderSystem/Nodes/ModifierNodes/RoundedCornersNode.swift (hunk 1 above)
Comment (johnny-duo) [reply Sources/Private/MainThread/NodeRenderSystem/Nodes/ModifierNodes/RoundedCornersNode.swift:5]: I don't use it much for work-related stuff but gave you a follow!

**Pull request merged**